types/base.d.ts

```diff
@@ -121,12 +121,12 @@ export interface ContextualCallbackTestContext extends CallbackTestContext {
 }
 
 export interface Macro<T, I, E> {
-	(t: T, input: I, expected: E): void;
-	title? (providedTitle: string, input: I, expected: E): string;
+	(t: T, input?: I, expected?: E): void;
```

Comment:
These should not be marked optional here, since that would require a user to make them optional as well. Making them optional in `test` and the generator script is sufficient. (A function with less arguments is assignable to a function type with more arguments.)